File: apps/app/lib/features/categories/domain/entities/category.dart
import 'package:equatable/equatable.dart';

/// A top-level catalogue category (`categories/{id}`).
///
/// Sub-categories are their own documents in a subcollection; the category doc
/// keeps a denormalised list of their names under `subCategoryNames`, exposed
/// here as [subCategories]. `productCount` is maintained server-side.
class Category extends Equatable {
  final String id;
  final String name;
  final String image;

  /// Denormalised sub-category names (Firestore key: `subCategoryNames`).
  final List<String> subCategories;
  final int productCount;
  final DateTime? createdAt;

  const Category({
    required this.id,
    required this.name,
    required this.image,
    required this.subCategories,
    required this.productCount,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        image,
        subCategories,
        productCount,
        createdAt,
      ];
}

File: packages/shared/src/types/category.ts
import type { Timestamp } from 'firebase/firestore';

/**
 * CategoryProps — a top-level catalogue category (spec §1.4).
 *
 * WHY here: shared by admin (CRUD) and storefronts (browse). Sub-categories are their OWN documents in
 * a subcollection (`categories/{id}/subCategories`, see SubCategoryProps); `subCategoryNames` is a
 * denormalized list of their names kept on the category doc so the Categories list can render the
 * "Sub-categories" column without reading the subcollection. `productCount` is maintained server-side.
 * Date fields use `Timestamp`.
 */
export interface CategoryProps {
  id: string;
  name: string;
  image: string;
  subCategoryNames: string[];
  productCount: number;
  keywords: string[];
  createdAt: Timestamp;
  updatedAt: Timestamp;
}
